ambari-server/src/main/java/org/apache/ambari/server/topology/Component.java

```diff
@@ -54,12 +59,20 @@ public String getName() {
   }
 
   /**
-   * @return the mpack associated with this component (can be {@code null} if component -> mpack mapping is unambiguous)
+   * @return the mpack associated with this component as {@link String} (can be {@code null} if component -> mpack mapping is unambiguous)
```

Comment:
Minor issue in the javadocs: 

Wouldn't this method return null if the mpack mapping was ambiguous, rather than unambiguous?